### ISSUE TITLE
Document rationale for using Project ID randomization

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-project-factory#723]
### TL;DR

Project Factory includes a feature that adds a randomized suffix to Project IDs, but does not clarify why someone would or would not want to do this

### Detailed design

Document when someone should choose to use Project ID randomization or not.

### Additional information

The [Project Factory API documentation](https://registry.terraform.io/modules/terraform-google-modules/project-factory/google/latest) has the following input: `random_project_id` with description: "Adds a suffix of 4 random characters to the project_id" and a default value of `false`. However, it does not indicate why and how someone would want to choose this setting.

This issue: https://github.com/terraform-google-modules/terraform-google-project-factory/issues/347 says that the random ID is useful for testing, since we can create and shut down projects without worrying about conflicts. Since projects can be shut down but require 30 days to delete (i.e. cannot be deleted immediately), the random suffix makes it possible to replace projects, since otherwise, you'd have to manually add a suffix (i.e. to delete `project-a`, you'd shut it down, then create a new one `project-a-2` since `project-a` would be pending deletion for 30 days)

Randomization may also have benefits because the Project ID is considered public and thus should not contain sensitive data, so it can be used to hide project details (i.e. set the text description to something meaningful, with an ID of something random like corp-proj-xxxxx)

**Comment by bharathkkb** [comment]
Hi @jawnsy 
Happy to add this to [docs](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/docs/FAQ.md). In addition to the points you mentioned, project IDs are also a global resource, so adding a random suffix helps prevent collisions. Also I believe Project IDs are actually never reused (30 days is the soft delete period when you can still restore the deleted project). [Docs](https://cloud.google.com/resource-manager/docs/creating-managing-projects) state `Cannot be in use or previously used; this includes deleted projects.`
